### PR TITLE
iOS Module + Example support

### PR DESCRIPTION
**Pull request by shreyasthiagaraj**
This PR adds an iOS React Native module. The module is written in Swift with proper Objective-C bridging. It lays the groundwork for quickly adding many more methods from the Blockstack SDK for React Native.

The included sample app will work by just dropping the three primary files into your project. It adds the following methods:

- `signIn`
- `signUserOut`
- `isUserSignedIn`
- `createSession`
- `hasSession`
- `putFile`
- `getFile`

NOTE: In this PR, the web app we point to has been modified to have a proper redirect.html file for auth. This may be indicative of a discrepancy between iOS and Android in consuming the redirect URI after auth as the previous web app did not work (this new one was taken from the iOS SDK sample).

### CODE REVIEW

File: example-sdk-module/App.js [hunk 1]
```diff
@@ -138,7 +139,7 @@ export default class App extends Component<Props> {
     result = await RNBlockstackSdk.signUserOut()
 
     console.log(JSON.stringify(result))
-    if (result["signedOut"]) {
+    if (result["ed"]) {
```
[the remark this answers is not in the document]
Comment (shreyasthiagaraj) [reply example-sdk-module/App.js]: Oops--this was a typo. Must have happened accidentally. Good catch :).

File: example-sdk-module/App.js [hunk 2]
```diff
@@ -147,7 +148,9 @@ export default class App extends Component<Props> {
     this.setState({fileUrl: "uploading..."})
     content = "Hello React Native"
     options = {encrypt: false}
+    console.log("\n\n\n\n\nPUT FILE START\n\n\n\n")
```
[the remark this answers is not in the document]
Comment (shreyasthiagaraj) [reply example-sdk-module/App.js]: Yep--was using this during debugging an issue with putfile/getfile. Dont need it anymore.

File: example-sdk-module/ios/Podfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  'DevSupport', # Include this to enable In-App Devmenu if RN >= 0.43
+  'RCTText',
+  'RCTNetwork',
+  'RCTWebSocket', # Needed for debugging
+  # Add any other subspecs you want to use in your project
+  ]
+  # Explicitly include Yoga if you are using RN >= 0.42.0
+  pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'
+
+  # Third party deps podspec link
```
[the remark this answers is not in the document]
Comment (shreyasthiagaraj) [reply example-sdk-module/ios/Podfile]: Yes. The FB tutorial has them too, along with answers I found around the web for issues I was having getting it to work.

**Comment by shreyasthiagaraj** [comment]
@friedger I've moved the code over into BlockstackSDK.

Currently there is a discrepency between how Android and iOS handles the auth response, which might explain why the provided web app in this project no longer works for Android. This is tracked here:
https://github.com/blockstack/blockstack-android/issues/125

The same data is used between both Android and iOS for the supported methods. The only discrepancy is signIn, which in Android provides just `decentralizedId` and `loaded` values, but in iOS also provides whether the operation succeeded, failed, or was cancelled by the user. I'm not sure if this information is available via Android's sdk, and I'm also having some issues with Android Studio to make this available. Either way though, it's just an additional data point in iOS that should be ignored for now.  If needed I can remove that to match Android. In the future there should probably be some document that syncs available data between the two platforms.

**Comment by shreyasthiagaraj** [comment]
@friedger bump. Thanks :)

**Comment by yknl** [comment]
Thanks for this @shreyasthiagaraj. I'm fine with merging this so that we can use it in React Native.

However, since blockstack.js version 19.0.0 was just released, and the library does not depend on a web browser environment now. Would it make more sense to modify that to get it to work on React Native?

In it's current form the RN lib wraps blockstack.js with Swift and then wraps that again with Javascript.

**Comment by hstove** [comment]
This would likely require a new StorageStrategy in Blockstack.js to not use `localStorage`. However, this new strategy could likely be the same as a node.js version.

**Comment by shreyasthiagaraj** [comment]
I'm not sure--I'm not too familiar with JS/node.js to comment on whether that could work. @hstove you're suggesting that building a node.js compatible version of Blockstack.js will make it work for React Native too?

**Comment by friedger** [comment]
@hstove @yknl Blockstack.js still depends on Crypto which was the problem why a swift and Android implementation was needed, wasn't it?
[AsyncStorage](https://facebook.github.io/react-native/docs/AsyncStorage.html) could be used for the storage strategy..

**Pull request merged**